File: poin-b/c++/Tim.cpp
#include <iostream>
#include <string>

/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * *                                                                                                                                           
 *  Saya Yosafat (2009929) mengerjakan evaluasi Tugas Praktikum dalam mata kuliah Desain dan Pemrograman Berorientasi Objek untuk          *
 *  keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.                                        *
 * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */

using namespace std;

// deklarasi kelas tim
class Tim{
    
    private:

        // deklarasi atribut
        string namaTim[2];
        string negaraAsal;
        string tahunBerdiri;
        string pemain[11];

    public:

        // deklarasi konstruktor
        Tim(string inputNamaTim[2], string inputNegaraAsal, string inputTahunBerdiri, string inputPemain[11]){
            for (int i = 0; i < 2; i++)
            {
                this->namaTim[i] = inputNamaTim[i];
            }
            
            this->negaraAsal = inputNegaraAsal;
            this->tahunBerdiri = inputTahunBerdiri;
            
            for (int i = 0; i < 11; i++)
            {
                this->pemain[i] = inputPemain[i];
            }  
        }

        // deklarasi getter sebagai method Read Only
        string getTim(int temp){

            return this->namaTim[temp];        
        }
        
        string getNegara(){
            return this->negaraAsal;
        }
        string getTahun(){
            return this->tahunBerdiri;
        }
        string getPemain(int temp){

            return this->pemain[temp];
        }

        // deklarasi setter sebagai method Write
        void setTim(string inputTim[2]){
            for (int i = 0; i < 2; i++)
            {
                this->namaTim[i] = inputTim[i];
            }
        }
        void setNegara(string inputNegara){
            this->negaraAsal = inputNegara;
        }
        void setTahun(string inputTahun){
            this->tahunBerdiri = inputTahun;
        }
        void setPemain(string inputPemain[11]){
            for (int i = 0; i < 11; i++)
            {
                this->pemain[i] = inputPemain[i];
            }            
        }

        // deklarasi method display untuk menampilkan data menggunakan method getter
        void display(){

            cout << "\n===== DATA TIM =====" << endl;
            
            cout << "Nama Tim      : ";

            for (int i = 0; i < 2; i++)
            {
                cout << this->getTim(i) << " ";
            }

            cout << endl;
            
            cout << "Negara Asal   : " << this->getNegara() << endl;
            cout << "Tahun Berdiri : " << this->getTahun() << endl;
            cout << "Pemain        : " << endl;
            
            int index = 1;

            for (int i = 0; i < 11; i++)
            {
                cout << index << ". " << this->getPemain(i) << endl;
                index++;
            }

            cout << "\nDATA BERHASIL DISIMPAN :D\n" << endl;
            
        }

        // deklarasi destruktor
        ~Tim(){
        }

};

File: poin-b/java/Tim.java
public class Tim {

    // deklarasi atribut private
    private String[] namaTim = new String[2];
    private String negaraAsal;
    private String tahunBerdiri;
    private String[] pemain = new String[11];

    
    // deklarasi konstruktor
    public Tim(String[] inputNamaTim, String inputNegaraAsal, String inputTahunBerdiri, String[] inputPemain){
        
        for (int i = 0; i < 2; i++) {
            this.namaTim[i] = inputNamaTim[i];
        }
    
        this.negaraAsal = inputNegaraAsal;
        this.tahunBerdiri = inputTahunBerdiri;

        for (int i = 0; i < 11; i++) {
            this.pemain[i] = inputPemain[i];
        }
    }

    // deklarasi getter sebagai method Read Only
    public String getTim(int temp){
        return this.namaTim[temp];
    }
    public String getNegara(){
        return this.negaraAsal;
    }
    public String getTahun(){
        return this.tahunBerdiri;
    }
    public String getPemain(int temp){
        return this.pemain[temp];
    }

    
    // deklarasi setter sebagai method Write
    public void setTim(String[] namaTim){

        for (int i = 0; i < 2; i++) {
            this.namaTim[i] = namaTim[i];
        }
    }
    public void setNegara(String negaraAsal){
        this.negaraAsal = negaraAsal;
    }
    public void setTahun(String tahunBerdiri){
        this.tahunBerdiri = tahunBerdiri;
    }
    public void setPemain(String[] pemain){
                
        for (int i = 0; i < 11; i++) {
            this.pemain[i] = pemain[i];
        }
    }

    // deklarasi method display untuk menampilkan data menggunakan method getter
    public void display(){
        
        System.out.println("\n===== DATA TIM =====");
        
        // tampilan untuk nama tim
        System.out.println("Nama Tim      : ");

        for (int i = 0; i < 2; i++) {
            System.out.println(this.getTim(i));
        }

        // tampilan untuk negara asal dan tahun berdiri tim
        System.out.println("Negara Asal   : " + this.getNegara());
        System.out.println("Tahun Berdiri : " + this.getTahun());

        // tampilan untuk pemain
        System.out.println("Pemain        : ");

        int index = 1;
        for (int i = 0; i < 11; i++) {
            System.out.print(index);
            System.out.print(". ");
            System.out.println(this.getPemain(i));
            index++;
        }

        System.out.println("\nDATA BERHASIL DISIMPAN :D\n");
    }

}
    